### PR TITLE
Windows tests

### PR DESCRIPTION
**Pull request by blakebyrnes**
This PR adds automated tests for windows, and changes the way we cleanup after tests to try to handle more scenarios where things blow up. As a part of this PR, the servers that we use that are only exposed internally are now seeking out their own open ports by default

### CODE REVIEW

File: core/lib/GlobalPool.ts [hunk 1]
```diff
@@ -37,9 +32,6 @@ export default class GlobalPool {
   }[] = [];
 
   public static async start() {
-    if (this.isStarted) {
-      throw new Error('GlobalPool is already started');
-    }
```
Comment (calebjclark) [inline core/lib/GlobalPool.ts:42]: Why get rid of `isStarted`?

Comment (blakebyrnes) [reply core/lib/GlobalPool.ts:42]: We weren't using it anymore. GlobalPool no longer had a single startup requirement in the code. I don't know when we switched to booting based on if a chromeCore instance existed or not, but that's what the code does. I think this is legacy from Ulixee. 

I was tracking down an issue in Windows where it was still shutting down during the next startup, and cleared this out when I saw it didn't actually do anything anymore. We never set isStarted to true.

File: core/lib/GlobalPool.ts [hunk 2]
```diff
@@ -90,17 +82,12 @@ export default class GlobalPool {
...
     }
-    if (this.chromeCore) {
-      await this.chromeCore.close();
-    }
-    if (this.mitmServer) {
-      await this.mitmServer.close();
-    }
-    SessionsDb.shutdown();
-    this.waitingForAvailability = [];
+    this.waitingForAvailability.length = 0;
```
Comment (calebjclark) [inline core/lib/GlobalPool.ts:85]: Why this change to `length = 0`?

Comment (blakebyrnes) [reply core/lib/GlobalPool.ts:85]: It's probably actually unnecessary. I was just keeping the same instance instead of potentially creating a new one in between "awaits" for closing.

File: core/lib/GlobalPool.ts (hunk 1 above)
Comment (calebjclark) [reply core/lib/GlobalPool.ts:42]: Ok, yeah... crazy that it was doing nothing.
